### ISSUE TITLE
List of missing arguments not displayed in D417 Missing argument description error

### ISSUE THREAD

**Issue by martinResearch** [PyCQA/flake8-docstrings#14]

**Comment by sigmavirus24** [comment]
I transferred it `flake8-docstrings` for you

**Comment by asottile** [comment]
it looks like `.message` includes that context but has a different problem -- it adds an extra `:` in the message after the code (which makes it incompatible with `flake8`)

two ideas there:

1. replace out the colon (seems a little fragile)
2. reimplement `pydocstyle`'s additional message logic (also seems kinda fragile)

**Comment by martinResearch** [comment]
I am not sure to understand why option 1 would be fragile. What do you think could break this solution? If flake8 does not handle error messages with ":" then it seems sensible to replace these, at the cost of denaturing some error messages. Alternatively, could flake8 be modified to handle error messages with ":" ?

**Comment by asottile** [comment]
pydocstyle has changed the format twice now looking at the git history of `flake8-docstrings` and I see no reason it would change again

changing flake8 is kinda out of the question -- a single plugin doing it wrong shouldn't loosen the framework

**Comment by martinResearch** [comment]
"a single plugin doing it wrong" not sure to have enough context to understand why using a ":" in the error message is wrong. Is it wrong independently of flake8's requirements? If so, is it documented anywhere that error messages should not use ":" outside the flake8 context? Or is that wrong only in the sense that it is not supported by flake8 and no other flake8 plug-in do that ? If so, it seems to me like it would be an improvement of flake8 to support it, and could potentially benefit other plugins.

**Comment by asottile** [comment]
yes, the documented message format is code then space then message

changing that would break lots of things

**Comment by s0undt3ch** [comment]
If the format is `[code] [message]`, then that's easy to parse, and message could virtually contain anything, so why restrain it?

**Comment by s0undt3ch** [comment]
In my case, pydocstyle reports
```
src/pytestshellutils/shell.py:327 in public method `run`:
        D417: Missing argument descriptions in the docstring (argument(s) cwd, env are missing descriptions in 'run' docstring)
```
While flake8 ...
```
flake8...............................................................................Failed
- hook id: flake8
- exit code: 1

src/pytestshellutils/shell.py:327:1: D417 Missing argument descriptions in the docstring
```

**Comment by s0undt3ch** [comment]
Ok, so, it's explicitly only using [shor_desc](https://github.com/PyCQA/flake8-docstrings/blob/1.6.0/flake8_docstrings.py#L145) .... Why not the [message](https://github.com/PyCQA/pydocstyle/blob/6.1.1/src/pydocstyle/violations.py#L52) attribute?

In my case, that seems to be the root cause for now showing what arguments are missing

**Comment by asottile** [comment]
@s0undt3ch I appreciate you trying to help but we've already [come to that above](https://github.com/PyCQA/flake8-docstrings/issues/14#issuecomment-1065352885) and have a reason why `message` is problematic

**Comment by s0undt3ch** [comment]
Is the `:` problematic because of programmatically parsing flake8 output?
Anyway, don't know the whole code base to better form an opinion/solution.

**Comment by s0undt3ch** [comment]
We should create(yet another) open standard for linter messages 🤣

**Comment by asottile** [comment]
I mean there is flake8's standard which is `CODE ASCII SPACE MESSAGE` that's why pydocstyle's (`.message`) `CODE COLON SPACE MESSAGE` does not work -- that's the whole point of the issue
